src/main/java/com/ibm/watson/developer_cloud/service/WatsonService.java

```diff
@@ -140,12 +140,19 @@ private Call createCall(Request request) {
       builder.url(RequestUtils.replaceEndPoint(request.url().toString(), getEndPoint()));
     }
 
+    String userAgent = getUserAgent();
+
     if (defaultHeaders != null) {
-      for (String key : defaultHeaders.names())
+      for (String key : defaultHeaders.names()) {
         builder.header(key, defaultHeaders.get(key));
+      }
+      if (defaultHeaders.get(HttpHeaders.USER_AGENT) != null) {
+        userAgent += "; " + defaultHeaders.get(HttpHeaders.USER_AGENT);
```

Comment:
Referring to [common browser user agents](https://en.wikipedia.org/wiki/User_agent), I'd suggest appending the custom user agent without semicolon, so that the final String can look like `watson-apis-java-sdk/3.0.0-RC1 (java.vendor=IBM Corporation; os.arch=x86_64; os.version=10.11.4; os.name=Mac OS X; java.version=1.8.0_66) my-application/1.0`